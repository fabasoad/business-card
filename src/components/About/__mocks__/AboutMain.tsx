import * as React from 'react'

export default function AboutMain() {
  return (
    <div data-testid="AboutMain"></div>
  )
}
