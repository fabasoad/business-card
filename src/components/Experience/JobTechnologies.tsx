import * as React from 'react'
import { useTranslation } from 'react-i18next'
import { v4 as uuidv4 } from 'uuid'
import TechnologyStorage from '../../scripts/technologies/TechnologyStorage'

interface JobTechnologiesProps {
  keys: string[]
}

export default function JobTechnologies(props: JobTechnologiesProps) {
  const technologyStorage = new TechnologyStorage()
  const { t } = useTranslation()
  return (
    <div className="technologies">
      <h5>{t('business-card-experience-technologies-title')}</h5>
      {props.keys
        .map((key) => technologyStorage.findByName(key))
        .map((t) => <img key={uuidv4()} src={t.img} alt={t.name} title={t.title} />)}
    </div>
  )
}
