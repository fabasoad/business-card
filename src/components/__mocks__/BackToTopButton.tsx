import * as React from 'react'

export default function BackToTopButton() {
  return (
    <div data-testid="BackToTopButton"></div>
  )
}
