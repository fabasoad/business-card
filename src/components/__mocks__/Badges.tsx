import * as React from 'react'

export default function Badges() {
  return (
    <div data-testid="Badges"></div>
  )
}
