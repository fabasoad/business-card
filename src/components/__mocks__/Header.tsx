import * as React from 'react'

export default function Header() {
  return (
    <div data-testid="Header"></div>
  )
}
