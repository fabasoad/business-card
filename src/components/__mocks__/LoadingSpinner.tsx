import * as React from 'react'
import { LoadingSpinnerProps } from '../LoadingSpinner'

export default function LoadingSpinner({ background, color }: LoadingSpinnerProps) {
  return (
    <div data-testid={`LoadingSpinner-${background}-${color}`}></div>
  )
}
