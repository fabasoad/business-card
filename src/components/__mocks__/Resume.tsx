import * as React from 'react'

export default function Resume() {
  return (
    <div data-testid="Resume"></div>
  )
}
