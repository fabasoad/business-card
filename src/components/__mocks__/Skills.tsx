import * as React from 'react'

export default function Skills() {
  return (
    <div data-testid="Skills"></div>
  )
}
