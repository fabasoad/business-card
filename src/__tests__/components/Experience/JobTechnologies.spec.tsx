/// <reference types="jest" />
import * as React from 'react'
import { shallow } from 'enzyme'
import JobTechnologies from '../../../components/Experience/JobTechnologies'
import { useTranslation } from '../../__mocks__/react-i18next'

let tMock

beforeEach(() => {
  tMock = useTranslation().t
})

test('should render JobTechnologies correctly', () => {
  const wrapper = shallow(<JobTechnologies keys={['maven', 'java']} />)
  expect(wrapper).toMatchSnapshot()
  expect(tMock).toBeCalledWith('business-card-experience-technologies-title')
})
