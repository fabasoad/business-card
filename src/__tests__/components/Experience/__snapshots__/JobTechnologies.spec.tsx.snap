// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render JobTechnologies correctly 1`] = `
<div
  className="technologies"
>
  <h5>
    business-card-experience-technologies-title
  </h5>
  <img
    alt="maven"
    key="test-uuid"
    src="test-file-stub"
    title="Maven"
  />
  <img
    alt="java"
    key="test-uuid"
    src="test-file-stub"
    title="Java"
  />
</div>
`;
